src/main/java/com/melon/portfoliomanager/services/CompanyStocksManager.java

```diff
@@ -2,10 +2,34 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 public class CompanyStocksManager {
+    private final double PRICE_CHANGE_PERCENT_THRESHOLD = 5;
     private final Map<String, Double> companyStocks = new HashMap<>();
 
+    public Map<String, Double> getCompanyStocksWithExtremePriceChanges(Map<String, Double> latestCompanyStocks) {
```

Comment:
What do you think about moving this logic as part of updateCompanyStockPrices? Realistically the only time we need to look for price changes is when we update prices. We can also call the Kafka service from here (if the returned list is not empty) instead of in StockPricesService. So the flow would be:

StockPricesService: on a schedule fetch price data -> store the results by calling CompanyStocksManager
CompanyStocksManager: store data, check for large price changes; if such are detected, publish a message in Kafka